src/sys/derivs.cxx

```diff
@@ -276,26 +276,50 @@ const Field2D D4DZ4(const Field2D &f, CELL_LOC outloc, const std::string &method
 /*!
  * Mixed derivative in X and Y
  *
- * This first takes derivatives in X, then in Y.
+ * This first takes derivatives in Y, then in X.
  *
- * ** Applies Neumann boundary in Y, communicates
+ * ** Communicates and applies boundary in X.
  */
-const Field2D D2DXDY(const Field2D &f, CELL_LOC outloc, const std::string &method, REGION region) {
-  Field2D dfdy = DDY(f, outloc, method, RGN_NOY);
+const Field2D D2DXDY(const Field2D& f, CELL_LOC outloc, const std::string& method,
+    REGION region, const std::string& dfdy_boundary_condition) {
+
+  // If staggering in x, take y-derivative at f's location.
+  const auto y_location =
+    (outloc == CELL_XLOW or f.getLocation() == CELL_XLOW) ? CELL_DEFAULT : outloc;
+
+  Field2D dfdy = DDY(f, y_location, method, region);
+
+  // Set x-guard cells and x-boundary cells before calculating DDX
   f.getMesh()->communicate(dfdy);
+  dfdy.applyBoundary("core", dfdy_boundary_condition);
+  dfdy.applyBoundary("pf", dfdy_boundary_condition);
+  dfdy.applyBoundary("sol", dfdy_boundary_condition);
```

Comment:
Is it possible for these to just be `applyBoundary(const std::string &condition)`? Seeing explicit boundary regions makes me nervous!

Reply:
They can be just `dfdy.applyBoundary(dfdy_boundary_condition)`. It's slightly sub-optimal as we don't need x-boundary conditions, but I think you're right, it's better to be safe if the names of regions change or get extended. Would be nice to have some way of applying a y-boundary condition that includes all regions - not sure if it's practical, but maybe we could think about it in the great boundary re-write?
Anyway, I'll change this and the one below...

Reply:
Yes, some way of specifying that would be nice and possibly even makes sense!